### PR TITLE
https://trello.com/c/YnrxYltC/158-transactions

### PR DESCRIPTION
**Pull request by Verbalist**
implement api endpoint for trs

### CODE REVIEW

File: api/controller/transaction/index.ts [hunk 1]
```diff
@@ -1,59 +1,71 @@
...
+    createTransaction(message: Message2<{ secret: string, trs: TransactionModel<IAsset> }>, socket: any) {
         SocketMiddleware.emitToCore(message, socket);
     }
 
+    @RPC(API_ACTION_TYPES.GET_TRANSACTION)
+    async getTransaction(message: Message2<{ id: string }>, socket: any) {
+        SocketMiddleware.emitToClient<TransactionApi<IAsset>>(
+            message.headers.id,
+            API_ACTION_TYPES.GET_TRANSACTIONS,
+            await TransactionPGRepository.getOne(message.body.id) as any,
```
Comment (macaqueee) [inline api/controller/transaction/index.ts]: you need to wrap data with `ResponseEntity`

File: api/controller/transaction/types.ts [hunk 2]
```diff
@@ -1,13 +1,10 @@
...
-import { TransactionModel } from 'shared/model/transaction';
+import { TransactionType } from 'shared/model/transaction';
+import { Sort } from 'api/utils/common';
 
-export type Filter = {
-    offset: number,
-    limit: number,
-    sort?: string,
-    type?: TransactionType
+export type getTransactionsRequest = {
```
Comment (macaqueee) [inline api/controller/transaction/types.ts]: create Pagination type with `{limit: number, offset: number}`

File: shared/driver/socket/codes.ts [hunk 3]
```diff
@@ -15,8 +12,12 @@ export const GET_REFERRED_USERS_REWARD = 'GET_REFERRED_USERS_REWARDS';
 
 export const GET_CURRENT_ROUND = 'GET_CURRENT_ROUND';
 
-export const CREATE_TRANSACTION = 'CREATE_TRANSACTION';
-export const GET_TRANSACTION = 'GET_TRANSACTION';
-export const GET_TRANSACTION_BY_ID = 'GET_TRANSACTION_BY_ID';
-export const GET_TRANSACTION_HISTORY = 'GET_TRANSACTION_HISTORY';
-export const GET_TRANSACTIONS_BY_BLOCK_ID = 'GET_TRANSACTIONS_BY_BLOCK_ID';
+export const enum API_ACTION_TYPES  {
```
Comment (axelsheva) [inline shared/driver/socket/codes.ts]: double space

File: api/controller/transaction/index.ts [hunk 4]
```diff
@@ -28,12 +28,10 @@ export class TransactionController {
...
     @RPC(API_ACTION_TYPES.GET_TRANSACTION)
     async getTransaction(message: Message2<{ id: string }>, socket: any): Promise<void> {
-        SocketMiddleware.emitToClient<ResponseEntity<TransactionModel<IAsset>>>(
+        SocketMiddleware.emitToClient<TransactionModel<IAsset>>(
             message.headers.id,
             message.code,
-            new ResponseEntity({
-                data: SharedTransactionRepo.serialize(await TransactionPGRepository.getOne(message.body.id))
-            }),
+            SharedTransactionRepo.serialize(await TransactionPGRepository.getOne(message.body.id)),
```
Comment (macaqueee) [inline api/controller/transaction/index.ts]: Maybe split to different variables for better readable?

**Pull request merged**